pages/employee/index.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+
+    <UForm :validate="validate" :state="employee" class="space-y-4" @submit="submitEmployee" @error="onError">
+        <div class="w-96">
+            <div class="mb-2">
+                <UFormGroup label="First Name" name="first_name">
+                    <UInput placeholder="Enter First Name" v-model="employee.first_name" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Last Name" name="last_name">
+                    <UInput placeholder="Enter Last Name" v-model="employee.last_name" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Mobile" name="mobile">
+                    <UInput placeholder="Enter Mobile" v-model="employee.mobile" :maxlength="10"
+                        @input="allowNumericValue" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Pincode" name="pincode">
+                    <UInput placeholder="Enter Pincode" :maxlength="6" v-model="employee.pincode" @change="checkPincode"
+                        @input="allowNumericValue" />
+                    <span v-if="pincodeError" class="text-red-500">{{ pincodeError }}</span>
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup name="Employment Type" label="Employment Type">
+                    <UInputMenu v-model="employee.employe_type" :options="employeeTypes" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup :label="incomeTitle" name="income">
+                    <UInput :placeholder="`Enter ${incomeTitle}`" v-model="employee.income" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UButton type="submit">
+                    Submit
+                </UButton>
+            </div>
+        </div>
+    </UForm>
+</template>
+
+<script setup lang="ts">
+import type { FormError, FormErrorEvent, FormSubmitEvent } from '#ui/types'
+const employeeTypes = ['Business', 'Salaried']
```

Comment:
Would recommend having an array of objects instead of straight strings as that would be a more accurate representation of API data we're used to.

Example:
`
const employeeTypes = [
{
    label: 'Business',
    value: 'business'
},
{
    label: 'Salaried',
    value: 'salaried'
},
]
`

Additionally, can be isolated in a `constants` folder for further use.

Reply:
Resolved